codesign/lib/src/utils.dart

```diff
@@ -5,10 +5,45 @@
 import 'dart:io';
 
 import 'package:args/args.dart';
+import 'package:file/file.dart';
 import 'package:process/process.dart';
 
 import 'log.dart';
 
+const String gsCloudBaseUrl = r'gs://flutter_infra_release';
+
+Future<void> upload({
+  required String localPath,
+  required String remotePath,
+  required String commitHash,
+  required ProcessManager processManager,
+}) async {
+  final String fullRemotePath = '$gsCloudBaseUrl/flutter/$commitHash/$remotePath';
+  final ProcessResult result = await processManager.run(
+    <String>['gsutil', 'cp', localPath, fullRemotePath],
+  );
+  if (result.exitCode != 0) {
+    throw Exception('Failed to upload $localPath to $fullRemotePath');
+  }
+}
+
+Future<File> download({
+  required String remotePath,
+  required String localPath,
+  required String commitHash,
+  required ProcessManager processManager,
+  required Directory tempDir,
+}) async {
+  final String source = '$gsCloudBaseUrl/flutter/$commitHash/$remotePath';
+  final ProcessResult result = await processManager.run(
+    <String>['gsutil', 'cp', source, localPath],
+  );
+  if (result.exitCode != 0) {
```

Comment:
Should this be updated to also verify the checksum matches what GCS has?

Reply:
umm seems like "gsutil automatically performs integrity checks on all uploads and downloads" based on the doc at https://cloud.google.com/storage/docs/gsutil/addlhelp/CRC32CandInstallingcrcmod. looks like hash verification is implemented as  https://cloud.google.com/storage/docs/gsutil/commands/cp#checksum-validation. looks like automatic retry has been implemented with https://cloud.google.com/storage/docs/gsutil/commands/cp#retry-handling. 

Looks like if retry of hash mismatch fail, gsutil will eventually throw a bad exit code, and we are catching the exit code in our code.